### DISCUSSION TITLE
Unable to load sound or album art on public page, behind nginx proxy manager.

### DISCUSSION THREAD

**Discussion by tvontheinternet** [AzuraCast/AzuraCast#4319] [Q&A]
I am using nginx reverse proxy manager [Nginx proxy manager](https://nginxproxymanager.com/) and have a proxy host setup there so that radio.my.site forwards to my station's public page. That took some doing to get pointed at the public page and not the management one, and the full conf is here: https://pastebin.com/raw/TpUcqWKp  (with lots of empty lines removed). The 
`location = /{return 301 $scheme://$http_host/public/urdak;}` line is what got it to not go to the management page:

The page loads, I can see the song that is playing and its progress but their is no sound or album art and it seems to be an SSL issue. Looking into the album art first hoping fixing that fixes the sound, using f12 in firefox > network > refresh the page to find the api request:

`GET https://192.168.0.176:4008/api/station/3/art/c2728c5bf7addcaba6d0b1f9-1624145202.jpg`

has a status of  "blocked". Clicking on that to load the above URL in a new tab I get a 

`Error code: SSL_ERROR_RX_RECORD_TOO_LONG` error. 

I can't find an error in azuoracast's logs, is there a way to get nginx errors from there vs access logs? (I am not sure if this issue is with nginx proxy manager or my azuoracast setup). When I run `docker-compose logs -f nginx_proxy` I can see the request for loading that image as:

`nginx_proxy                | nginx.1     | radio.my.site 192.168.0.130 - - [20/Jun/2021:06:47:06 +0000] "\x16\x03\x01\x02\x00\x01\x00\x01\xFC\x03\x03.E\xA4\xABd\x06:\xFAk|(\x9C\x1D\x1C\x91\xF8+.\x80\xED{\xCA3\x11\xAA\xDC2\xE1[\x1E\xFF\xED \x88\xCDq\x1C\x80\x86\xC0\xBA%\xA1\xC9e\x82\x9E/'R\x80\x00\xEDh2ev \xFC.\x0Fj\x88\x87t\x00$\x13\x01\x13\x03\x13\x02\xC0+\xC0/\xCC\xA9\xCC\xA8\xC0,\xC00\xC0" 400 150 "-" "-"`

I am using docker for azuracast, full config-yml here: https://pastebin.com/Q32sYh8j  

Looking for any advice on investigating this.

Edit: A little more info. I found the sound is not loading due to a NET::ERR_CERT_AUTHORITY_INVALID error. After selecting accept the risk and continue in firefox I can now load sound. Id like to fix this though permanently. 

I also noticed a private window in firefox after the above simply loads the page insecure and both sound and album art load. 
Https for the site as a whole does not throw any warnings though which seems odd? Just plying sound and the album art. 

In chrome, security overview I am getting:

Certificate - missing
This site is missing a valid, trusted certificate (net::ERR_CERT_AUTHORITY_INVALID).

Certificate - Subject Alternative Name missing
The certificate for this site does not contain a Subject Alternative Name extension containing a domain name or IP address.

This mesaage in the nginx_proxy_letsencrypt log may be of import?

`nginx_proxy_letsencrypt    | [Sun Jun 20 07:29:49 UTC 2021] radio.my.site:Verify error:Invalid response from http://radio.tvent.cool/.well-known/acme-challenge/hUoXlBZw6fK7X3avn3U1T3_osrJqgDoMC7PcAP0sGQA [my public IP]: 
`

Edit again: leaving this but I bet its because I can't read and am trying to use non standard ports since my goal was to run this on my machine running other docker containers and 80/443 was in use: 

_AzuraCast's web server must be served on the default ports, 80 for HTTP and 443 for HTTPS. By default, AzuraCast is already set up this way, but if you've modified the ports to serve the site on a secondary port, you must switch the ports back to the defaults when setting up LetsEncrypt and when performing renewals._

edit edit AGAIN, apologies but don't want to leave info out: Brought up a whole new instance on a different VM with a fresh docker install using default ports and got the same Verify error, invalid response as pasted above. 

**so this seems to be more simple than above and tl;dr, getting the following lets encrypt error on a fresh install: **

`nginx_proxy_letsencrypt    | [Sun Jun 20 08:21:46 UTC 2021] radio.my.site:Verify error:Invalid response from http://radio.my.site/.well-known/acme-challenge/7OX_WFzCM30MKJkoTvkJ28dl4NEpHaeydDTv4QWMSv0 [my public IP]: 
nginx_proxy_letsencrypt    | [Sun Jun 20 08:21:46 UTC 2021] Please check log file for more details: /dev/null
`

**Comment by ItsMitchh** [comment]
Greetings, I propose that you turn this into a issue post as it might assist you further than our Discussion pages as this may be a legitimate issue.

**Reply by tvontheinternet** [reply]
Thanks, will revisit in a bit as well working on this I hit lets encrypts weekly limit on requests
